dkg-gadget/src/worker.rs

```diff
@@ -588,6 +609,58 @@ where
 		}
 	}
 
+	fn verify_signature_against_authorities(
+		&mut self,
+		signed_dkg_msg: SignedDKGMessage<Public, DKGPayloadKey>
+	) -> Result<DKGMessage<Public, DKGPayloadKey>, String> {
+		let dkg_msg = signed_dkg_msg.msg;
+		let encoded = dkg_msg.encode();
+		let signature = signed_dkg_msg.signature.unwrap_or_default();
+		let can_proceed = {
+			// Get authority accounts
+			let mut authority_accounts: Option<(Vec<AccountId32>, Vec<AccountId32>)> = None;
+
+			if let Some(header) = self.latest_header.as_ref() {
+				let at = BlockId::hash(header.hash());
+				let accounts = self.client.runtime_api().get_authority_accounts(&at).ok();
+
+				if accounts.is_some() {
+					authority_accounts = accounts;
+				}
+			}
+
+			if authority_accounts.is_none() {
+				return Err("No authorities".into());
+			}
+
+			let check_signers = |xs: Vec<AccountId32>| {
+				return dkg_runtime_primitives::utils::verify_signer_from_set(
+					xs.iter().map(|x| {
+						sr25519::Public(to_slice_32(&x.encode()).unwrap_or_else(|| {
+							panic!("Failed to convert account id to sr25519 public key")
+						}))
+					}).collect(),
+					&encoded,
+					&signature,
+				)
+				.1
+			};
+
+			if !check_signers(authority_accounts.clone().unwrap().0.into())
+				|| !check_signers(authority_accounts.clone().unwrap().1.into()) {
```

Comment:
Shouldn't it be && instead of || ?

Reply:
Yea this is fixed now, I want messages to be valid if they come from either current or next authorities. Double-check this looks good now @temld4 .